pkgs/development/compilers/purescript/purescript/default.nix

```
@@ -0,0 +1,53 @@
+{ pkgs ? import <nixpkgs> {} }:
+
+# from justinwoo/easy-purescript-nix
+# https://github.com/justinwoo/easy-purescript-nix/blob/d383972c82620a712ead4033db14110497bc2c9c/purs.nix
+
+let
+  dynamic-linker = pkgs.stdenv.cc.bintools.dynamicLinker;
+
+  patchelf = libPath :
+    if pkgs.stdenv.isDarwin
+      then ""
+      else
+        ''
+          chmod u+w $PURS
+          patchelf --interpreter ${dynamic-linker} --set-rpath ${libPath} $PURS
+          chmod u-w $PURS
+        '';
+
+in pkgs.stdenv.mkDerivation rec {
+  name = "purs-simple";
+  version = "v0.12.3";
+
+  src =
+    if pkgs.stdenv.isDarwin
+    then
+    pkgs.fetchurl {
+      url = "https://github.com/purescript/purescript/releases/download/v0.12.3/macos.tar.gz";
+      sha256 = "1f916gv4fz571l4jvr15xjnsvjyy4nljv2ii9njwlm7k6yr5m0qn";
+    }
+    else
+    pkgs.fetchurl {
+      url = "https://github.com/purescript/purescript/releases/download/v0.12.3/linux64.tar.gz";
+      sha256 = "1fad862a2sv4njxbbcfzibbi585m6is3ywb94nmjl8ax254baj3i";
+    };
+
+
+  buildInputs = [ pkgs.zlib
+                  pkgs.gmp
+                  pkgs.ncurses5];
+  libPath = pkgs.lib.makeLibraryPath buildInputs;
+  dontStrip = true;
+
+  installPhase = ''
+    mkdir -p $out/bin
+    PURS="$out/bin/purs"
+
+    install -D -m555 -T purs $PURS
+    ${patchelf libPath}
+
+    mkdir -p $out/etc/bash_completion.d/
+    $PURS --bash-completion-script $PURS > $out/etc/bash_completion.d/purs-completion.bash
+  '';
```

GitHub ate my comment! Man! Oh well, here we go again :D

This will need a `meta` attribute, like this:

```suggestion
  '';
    meta = with stdenv.lib; {
    description = "A strongly-typed functional programming language that compiles to JavaScript";
    homepage = http://www.purescript.org/;
    license = licenses.bsd3;
    maintainers = [ maintainers.justinwoo ];
    platforms = [ "x86_64-linux" "x86_64-darwin" ];
  };
```

For this, we'll also need you to add yourself to the maintainer list in a separate commit: https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix. Since  you're on GitHub, definitely fill in the `github` attribute, but since you didn't sign your commits with GPG skip the `keys` attribute:

```
  justinwoo = {
    email = "moomoowoo@gmail.com";
    name = "Justin Woo";
    github = "justinwoo";
  };
```